inabox/tests/setup_operator_harness.go

```diff
@@ -146,63 +142,155 @@ func (oh *OperatorHarness) StartOperators(ctx context.Context) error {
 		return fmt.Errorf("no operators found in config")
 	}
 
-	oh.logger.Info("Starting operator goroutines", "count", operatorCount)
+	logger.Info("Pre-creating listeners for operator goroutines", "count", operatorCount)
+
+	// Pre-create all listeners with port 0 (OS assigns ports)
+	allListeners := make([]operatorListeners, operatorCount)
+	for i := range operatorCount {
+		listeners, err := oh.createOperatorListeners(i)
+		if err != nil {
+			// Clean up any listeners we created before failing
+			oh.closeAllListeners(logger, allListeners[:i])
+			return err
+		}
+		allListeners[i] = listeners
+
+		logger.Info("Created listeners for operator", "index", i,
+			"v1DispersalPort", listeners.v1Dispersal.Addr().(*net.TCPAddr).Port,
+			"v1RetrievalPort", listeners.v1Retrieval.Addr().(*net.TCPAddr).Port,
+			"v2DispersalPort", listeners.v2Dispersal.Addr().(*net.TCPAddr).Port,
+			"v2RetrievalPort", listeners.v2Retrieval.Addr().(*net.TCPAddr).Port)
+	}
 
-	// Start each operator
-	for i := 0; i < operatorCount; i++ {
-		instance, err := oh.startOperator(ctx, i)
+	// Now start each operator with its pre-created listeners
+	for i, listeners := range allListeners {
+		server, serverV2, err := oh.startOperator(ctx, logger, i, listeners)
 		if err != nil {
-			// Clean up any operators we started before failing
-			oh.Cleanup(context.Background(), oh.logger)
+			// Clean up any operators we started and all remaining listeners
+			oh.Cleanup(context.Background(), logger)
+			oh.closeAllListeners(logger, allListeners[i:])
```

Comment:
Is the idea here that any listener already handled will be closed by `oh.Cleanup(context.Background(), logger)`?